src/Pages/Expenses.cshtml.cs

```diff
@@ -138,15 +139,13 @@ private async Task SetData()
             try
             {
                 UserId = int.Parse(User.FindFirst(ClaimTypes.Name).Value);
-                Expenses = await EntryDatabase.ReadListAsync(x => x.Date >= StartDate && x.Date <= EndDate,
-                    UserId,
-                    EntryType.Expense);
-
-                ExpensesToDisplay = Expenses.OrderByDescending(x => x.Date)
-                    .ToIEntryList()
-                    .GetPage(CurrentPage, PageSize);
-
-                AllExpenses = Expenses.GetSum();//Replaced with sum from database.
+                var entryList = await EntryDatabase.ReadListAsync(UserId, EntryType.Expense);
+                Expenses = entryList.GetFrom(StartDate).GetTo(EndDate);
+                var (expenseList, totalPages) = await EntryDatabase.ReadByPage(x => x.Date >= StartDate && x.Date <= EndDate, UserId,
+                    EntryType.Expense, CurrentPage, PageSize);
+                ExpensesToDisplay = expenseList;
+                TotalPages = totalPages;
+                AllExpenses = Expenses.GetSum();
```

Comment:
So this is the part where it collides, we have to read whole list to get sum, and then we read just a part of entries to display? :D

Reply:
Yeah, after I created this I realised that there is not much to this method... :D Because we read all the entries anyways to get the sum, with implementation like this it's actually a lot slower then it was before :D

Reply:
But there's has to be a workaround, since everytime to read 3000 entries to display 10 is illogical, maybe the sum then should be removed from display and we won't need it?

Reply:
We can do that, but then there has to be a place where user can see all this data :D